static_src/actions/user_actions.js

```diff
@@ -257,25 +253,27 @@ const userActions = {
       entityGuid
     });
 
-    return cfApi.putAssociateUserToEntity(userGuid, entityGuid, entityType)
+    if (entityType === ORG_NAME) {
+      entityGuid = orgGuid;
+      cfApiRequest = cfApi.putAssociateUserToOrganization.bind(cfApi, userGuid, entityGuid);
+    } else {
+      entityGuid = SpaceStore.currentSpaceGuid;
+      cfApiRequest = cfApi.putAssociateUserToSpace.bind(cfApi, userGuid, orgGuid, entityGuid);
+    }
+
+    return cfApiRequest()
       .then(() => userActions.fetchEntityUsers(entityGuid, entityType))
       .then(entityUsers => userActions.createdUserAndAssociated(userGuid, entityGuid, entityUsers));
   },
 
   fetchEntityUsers(entityGuid, entityType) {
-    let entityUsers;
-    if (entityType === ORG_NAME) {
-      entityUsers = cfApi.fetchOrgUsers(entityGuid);
-    } else {
-      entityUsers = cfApi.fetchSpaceUserRoles(entityGuid);
-    }
-    return Promise.resolve(entityUsers);
+    return cfApi[(entityType === ORG_NAME) ? 'fetchOrgUsers' : 'fetchSpaceUserRoles'](entityGuid);
```

Comment:
If we were doing more things in the branch, I would advocate to move the code into separate methods to maintain readability. I think this is ok as is, for now.